ln-gateway/src/lib.rs

```diff
@@ -219,21 +224,22 @@ impl LnGateway {
 
     async fn buy_preimage_internal(
         &self,
-        payment_params: &PaymentParameters,
+        payment_hash: &sha256::Hash,
+        invoice_amount: &Amount,
         mut rng: impl RngCore + CryptoRng,
     ) -> Result<[u8; 32]> {
         let (out_point, _) = self
             .federation_client
-            .buy_preimage_offer(
-                &payment_params.payment_hash,
-                &payment_params.invoice_amount,
-                &mut rng,
-            )
+            .buy_preimage_offer(payment_hash, invoice_amount, &mut rng)
             .await?;
+
+        debug!("Decrypting preimage {}", payment_hash);
```

Comment:
:+1: for adding logging, we should do that more and not only when SHTF

nit: should probably be "awaiting decryption of preimage of hash"

Reply:
💯